sources/api/netdog/src/net_config/devices/bonding.rs

```diff
@@ -0,0 +1,221 @@
+use super::{error, Dhcp4ConfigV1, Dhcp6ConfigV1, Result, RouteV1, StaticConfigV1, Validate};
+use super::{validate_addressing, HasDhcp, HasRoutes, HasStatic};
+use crate::interface_name::InterfaceName;
+use serde::de::Error;
+use serde::{Deserialize, Deserializer};
+use snafu::ensure;
+use std::net::IpAddr;
+
+// Since we are checking for kind in a custom Deserializer, we can't use annotations for fields
+#[derive(Debug)]
+pub(crate) struct NetBondV1 {
+    pub(crate) primary: Option<bool>,
+    pub(crate) dhcp4: Option<Dhcp4ConfigV1>,
+    pub(crate) dhcp6: Option<Dhcp6ConfigV1>,
+    pub(crate) static4: Option<StaticConfigV1>,
+    pub(crate) static6: Option<StaticConfigV1>,
+    pub(crate) routes: Option<Vec<RouteV1>>,
+    #[allow(dead_code)]
+    kind: String,
+    pub(crate) mode: BondMode,
+    pub(crate) min_links: Option<u32>,
+    pub(crate) monitoring_config: BondMonitoringConfig,
+    pub(crate) interfaces: Vec<InterfaceName>,
+}
+
+impl<'de> Deserialize<'de> for NetBondV1 {
```

Comment:
I'm not 100% convinced this is better than the TryFrom String bits, but this allows us to catch the specifics in line during deserialization. We might be able to improve the casting one struct as another too at the end of this function but wanted to get it out to review.

Reply:
@zmrow  helped me on the piece I was missing, this is now much cleaner and I'm happy with where it ended up.